spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/test/java/com/example/DatastoreBookshelfExampleIntegrationTests.java

```diff
@@ -41,55 +41,98 @@
 @ExtendWith(SpringExtension.class)
 @SpringBootTest(
     classes = DatastoreBookshelfExample.class,
-    properties = "spring.shell.interactive.enabled=false",
     webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @EnabledIfSystemProperty(named = "it.datastore", matches = "true")
 class DatastoreBookshelfExampleIntegrationTests {
 
-  @Autowired private Shell shell;
-
   @Autowired private DatastoreTemplate datastoreTemplate;
 
-  @Autowired private BookRepository bookRepository;
-
   @Autowired private TestRestTemplate restTemplate;
 
+  @BeforeEach
+  void saveBooks() {
+    sendRequest(
+        "/saveBook",
+        "{\"title\":\"The Moon Is a Harsh Mistress\", \"author\":\"Robert A. Heinlein\", \"year\":1966}",
+        HttpMethod.POST);
+    sendRequest(
+        "/saveBook",
+        "{\"title\":\"Stranger in a Strange Land\", \"author\":\"Robert A. Heinlein\", \"year\":1961}",
+        HttpMethod.POST);
+    sendRequest(
+        "/saveBook",
+        "{\"title\":\"The Crack in Space\", \"author\":\"Philip K. Dick\", \"year\":1966}",
+        HttpMethod.POST);
+    sendRequest(
+        "/saveBook",
+        "{\"title\":\"Ubik\", \"author\":\"Philip K. Dick\", \"year\":1969}",
+        HttpMethod.POST);
+  }
+
   @AfterEach
   void cleanUp() {
     this.datastoreTemplate.deleteAll(Book.class);
   }
 
   @Test
-  void testSerializedPage() {
-    Book book = new Book("Book1", "Author1", 2019);
-    book.id = 12345678L;
-    this.bookRepository.save(book);
-    Awaitility.await().atMost(15, TimeUnit.SECONDS).until(() -> this.bookRepository.count() == 1);
-    String responseBody = sendRequest("/allbooksserialized", null, HttpMethod.GET);
-    assertThat(responseBody).contains("content\":[{\"id\":12345678}],\"pageable\":");
-    assertThat(responseBody).containsPattern("\"urlSafeCursor\":\".+\"");
```

Comment:
Noticed you removed this test. Looks like this was intentional added as part of a fix in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/commit/e46e2c1ff07c2cbe127aeb3596a24d1566431a14. Can you make sure there are relevant integration test covering this instead?

Reply:
I restored this test with minor changes.